neherlab/covid19_scenarios
# moved persisting settings logic to redux store

## Related issues and PRs

 - Related to #272 

## Description

Moved the global settings to a redux reducer because settings are a global state and moving them to the redux store makes sure that redux is the single source of truth for getting and setting these values. I think that manually updating settings in localstorage directly could lead to some problems down the road, (like someone forgetting to update it) so I moved the handling of persisting the settings to localstorage as a side effect of updating the redux store.

## Impacted Areas in the application

Main.tsx
ResultsCard.tsx
src/state

## Testing

1) Open the site, Refresh the page, observe that your UI preference is restored.
1.1) Checks for keys not in localstorage
2) Open the site, toggle a UI setting, such as auto run, log graphs and show humanize. Refresh the page, observe that your UI preference is restored. 
2.1) Checks for keys in localstorage and values are updated
